torch/csrc/jit/passes/remove_mutation.cpp

```diff
@@ -139,15 +142,37 @@ void MutationRemover::RemoveListMutation(Block* block) {
 
     // We rewrite something like:
     // x = {v0}
-    // x.append(v1)
+    // x.append(v1) (or x.insert(0, v1))
     // to:
-    // x = {v0, v1}
+    // x = {v0, v1} (or x = {v1, v0})
     // We can remove x.append from the the alias db list of writes.
     // All other aliasing properties remain valid.
     Node* list_construct = mutated_value->node();
-    list_construct->addInput(node->inputs().at(1));
-    node->output()->replaceAllUsesWith(mutated_value);
-    aliasDb_->writeIndex_->erase(node);
+    switch (node->kind()) {
+      case aten::append:
+        list_construct->addInput(node->inputs().at(1));
+        break;
+      case aten::insert: {
+        auto pos = toIValue(node->inputs().at(1))->toInt();
+        // insert to neg position is the same as insert to 0
+        // insert beyond current list length is the same as append
+        pos = std::min(
+            static_cast<size_t>(std::max(pos, 0L)),
+            list_construct->inputs().size());
+        list_construct->insertInput(pos, node->inputs().at(2));
+        break;
+      }
+      default:
+        TORCH_INTERNAL_ASSERT(false);
+    }
+
+    // process use-chain and aliasing of node output
+    bool has_output = (node->outputs().size() > 0);
```

Comment:
just fyi, neither append nor insert return a value in python so it's basically impossible for the outputs to have uses if people are scripting normal python code. i submitted a PR to fix this but unfortunately onnx relies on it and it doesnt really hurt anything, just kind of confusing, so I didn't push to land the PR: https://github.com/pytorch/pytorch/pull/39536.

either way this is fine here for the potentially confused, even though i think it will never be called